### PR TITLE
Reverts DAL.

### PR DESCRIPTION
**Pull request by PJB3005**
I did some profiles.
# **DAL is MORE THAN TWICE as expensive as baylights.**

Relevant profiles here **(Both were power cycling the entire station 5 times.):**
- [DAL](http://pastebin.com/MFzPpzWc)
- [Baylights](http://pastebin.com/Ryi1hLM7)

I do not believe that the following should be taken as proof that baylights is worse, as this SHOULD be handled by **`scheck()`**; either `scheck()` is **broken** **OR** baylights _just_ got caught in the **crossfire** of the fucking **moronic panic writer.**

**EDIT: Here's what's causing the source of the spam of calls during "a bout of lag" in the following image.**

`update_lumcount()`, the proc you see a LOT in the above image, is NOT expensive, it is some basic mathematical operations to check if lighting overlays actually get a visual change.

It gets called so much because it is the proc used to add or remove lumens to or from a lighting overlay, if an `apply_lum()` or `remove_lum()` call happens during "a bout of lag", this is bound to happen, as these are the procs that update the effect of light sources.

However after these procs are done, `scheck()` will kick in and pause the lighting controller.

**TL;DR** baylights was in the wrong place at the wrong time.

Also see my comments down below.

### CODE REVIEW

**Comment by FudgePucker** [comment]
Wait so this means that the fancy baylighting wasn't lagging?
Help.
Whats DAL?
What direction is up?
What is the color blue?

**Comment by Exxion** [comment]
Wait a second
I thought DAL was what we had before you ported baylights
I'm roughly as confused as Fudge

I'd look at the changes to try to clear up my confusion, but there's no way in hell that page will ever load with that many changes

**Comment by PJB3005** [comment]
@FudgePucker DAL was the lighting system we had before I ported bay's system.

After numerous optimizations we got it not lagging, pomf reverted it on an **EXTREMELY** misguided basis.

**Comment by FudgePucker** [comment]
So that means that the really nice looking lighting with all its smoothness got removed because it lagged,and we reverted to the really ugly lighting that lags and looks really blocky?
Okay now it makes more sense.

**Comment by Fox-McCloud** [comment]
Even when DAL is 100% properly+correctly implemented, it's still roughly 10% more expensive than "Baylights"...I also did profiling and while I didn't come up with the double amount that @PJB3005 did, it was still higher than Baylights.

TG's lighting system, on the other hand is roughly ~60% cheaper than Baylights (but no pretty colors).

**Comment by PJB3005** [comment]
Me and @clusterfack did new profiling, we put 100 flares on a HUGE conveyor going around central primary and then profiled that for 5 minutes, on both the DAL and bay side, here are the results.
* [DAL](http://pastebin.com/urcHzCuX)
  * CPU usage was at ~11% average for the entire profiling. (from task manager)
* [Baylights](http://pastebin.com/riAiCeJE)
  * CPU usage was at ~9% average for the entire profiling. (from task manager)

**TL;DR** baylights is a little bit less than twice as fast.

Note that this was tested on a tick_lag of 0.3

**Comment by PsiOmegaDelta** [comment]
I have to assume then that the CPU usage without any light was at ~7%?

**Comment by PJB3005** [comment]
Uh, I didn't test that.

**Comment by PsiOmegaDelta** [comment]
Then I don't see where your conclusion about twice as fast comes from, using only those particular numbers.

**Comment by PJB3005** [comment]
I was basing it of the profiles I put in pastebins

**Comment by PJB3005** [comment]
Another source of lag is that we buffed light ranges for things like flashlights to accomodate for lambertian, they weren't lowered again after we switched back so that effected performance a lot in the negative side for baylights.

**Comment by Exxion** [comment]
But have you profiled in under normal usage scenarios? Baylights is definitely better in the worst case, but what about the average case?

**Comment by PJB3005** [comment]
average case it's worse because flashlights and shit are really goddamn stronk on baylights causing lag.

**Comment by clusterfack** [comment]
I don't know how to devise many more tests in which DAL can even win. The flares on a conveyor was, in my opinion, totally biased against the baylights system yet it still handled it flawlessly.

**Comment by PJB3005** [comment]
We determined opacity changes cause a HEAVY performance impact, I'm implementing VERY big performance improvements on this, thanks to a suggestion from @clusterfack.

**Comment by clusterfack** [comment]
I am confident that we have identified the weakness of bay lighting and forcibly removed the problem.

**Comment by nssatlantis** [comment]
Mind telling the rest of us what that precise problem was and how other sources could get rid of it? I'd just look into the commit but it's been squished into 1 so it's a bit of a search.

**Comment by clusterfack** [comment]
When opacity changes, or an opaque object is moved, every light source touching that changed turf will be 'force updated'. This means removing however many affecting light sources and then reapplying them, which as you can imagine is horrible.

Now the system was changed to recheck which turfs it is now affecting, then do modifications by comparing against the turfs it was applied to before. So instead of removing lum from say 30 turfs then applying lum to 32 turfs, we find which two turfs need to be changed and modify them directly.

**Comment by PJB3005** [comment]
goddamnit it doesn't compile.

**Comment by d3athrow** [comment]
BAN EVERYTHING

**Comment by clusterfack** [comment]
Re-merge when @d3athrow, re-merge when

**Comment by d3athrow** [comment]
what the fuck ever
